src/shared/install.c

```diff
@@ -138,10 +138,106 @@ static int mark_symlink_for_removal(
         return 0;
 }
 
+static int find_symlink_to_template(const char *root_dir, const char *config_path, const char *template, char **link_name) {
+        int r;
+        const char **l, *scope;
+        const char *vendor_lookup_paths[] = {
+                "/usr/lib/systemd",
+#ifdef HAVE_SPLIT_USR
+                "/lib/systemd",
+#endif
+                NULL
+        };
+
+        assert(config_path);
+        assert(link_name);
+        assert(unit_name_is_valid(template, UNIT_NAME_TEMPLATE));
+
+        scope = strstr(config_path, "user") ? "/user/" : "/system/";
+
+        STRV_FOREACH(l, vendor_lookup_paths) {
+                _cleanup_free_ char *path = NULL;
+                _cleanup_closedir_ DIR *d = NULL;
+                struct dirent *de;
+
+                path = strjoin(root_dir ?: "", *l, scope, NULL);
```

Comment:
Ah, I overlooked this one earlier. This can be handled more nicely with `strempty()`, but that can be fixed later.